bundles/org.openhab.binding.siemenshvac/src/main/java/org/openhab/binding/siemenshvac/internal/discovery/SiemenesHvacDiscoveryParticipant.java

```diff
@@ -0,0 +1,111 @@
+import org.osgi.service.component.annotations.Activate;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Modified;
+
+/**
+ * The {@link SiemensHvacDiscoveryParticipant} is responsible for discovering new and
+ * removed siemensHvac bridges. It uses the central {@link UpnpDiscoveryService}.
+ *
+ * @author Laurent Arnal - Initial contribution
+ */
+@NonNullByDefault
+@Component(configurationPid = "discovery.siemenshvac", immediate = true)
+public class SiemenesHvacDiscoveryParticipant implements UpnpDiscoveryParticipant {
+
+    @Activate
+    public void activate(@Nullable Map<String, Object> configProperties) {
+    }
+
+    @Modified
+    public void modified(@Nullable Map<String, Object> configProperties) {
+    }
+
+    @Override
+    public Set<ThingTypeUID> getSupportedThingTypeUIDs() {
+        return SiemensHvacBindingConstants.SUPPORTED_THING_TYPES;
+    }
+
+    @Override
+    public @Nullable DiscoveryResult createResult(RemoteDevice device) {
+        ThingUID uid = getThingUID(device);
+        if (uid != null) {
+            Map<String, Object> properties = new HashMap<>();
+            String ipAddress = device.getDetails().getPresentationURI().getHost();
+            properties.put(SiemensHvacBindingConstants.BASE_URL, "https://" + ipAddress + "/");
+
+            String label = "";
+
+            if (uid.getAsString().contains("ozw672")) {
+                label = "OZW672 IP Gateway";
+            } else if (uid.getAsString().contains("ozw772")) {
+                label = "OZW772 IP Gateway";
+            }
+
+            String serialNumber = device.getDetails().getSerialNumber();
+            DiscoveryResult result;
+            if (serialNumber != null && !serialNumber.isBlank()) {
+                properties.put(PROPERTY_SERIAL_NUMBER, serialNumber.toLowerCase());
+
+                result = DiscoveryResultBuilder.create(uid).withProperties(properties).withLabel(label)
+                        .withRepresentationProperty(PROPERTY_SERIAL_NUMBER).build();
+            } else {
+                result = DiscoveryResultBuilder.create(uid).withProperties(properties).withLabel(label).build();
+            }
+            return result;
+        } else {
+            return null;
+        }
+    }
+
+    @Override
+    public @Nullable ThingUID getThingUID(RemoteDevice device) {
+        DeviceDetails details = device.getDetails();
+        if (details != null) {
+            ModelDetails modelDetails = details.getModelDetails();
+            String serialNumber = details.getSerialNumber();
+            if (modelDetails != null && serialNumber != null && !serialNumber.isBlank()) {
+                String modelName = modelDetails.getModelName();
+                if (modelName != null) {
+                    if (modelName.startsWith("Web Server OZW672")) {
+                        return new ThingUID(SiemensHvacBindingConstants.THING_TYPE_OZW, "ozw672");
```

Comment:
`id` should be unique, for example serialNumber (if available always) or something that otherwise identifies the concrete bridge. Using a constant like this prevent you from adding more than one bridge.

See for example:

https://github.com/openhab/openhab-addons/blob/2bd4e11dd8d8a2f7c366a2d2b730117711e1c481/bundles/org.openhab.binding.danfossairunit/src/main/java/org/openhab/binding/danfossairunit/internal/discovery/DanfossAirUnitDiscoveryService.java#L142

or:

https://github.com/openhab/openhab-addons/blob/2bd4e11dd8d8a2f7c366a2d2b730117711e1c481/bundles/org.openhab.binding.hue/src/main/java/org/openhab/binding/hue/internal/discovery/HueBridgeMDNSDiscoveryParticipant.java#L158-L172

Reply:
Has I add the serialNumber to the thing ID, it should be ok now.
Fix in next commit